Client/WarFare/PlayerMySelf.cpp

```diff
@@ -777,23 +777,30 @@ bool CPlayerMySelf::IsAttackableTarget(CPlayerBase* pTarget, bool bMesureAngle)
 	return true;
 }
 
+// TODO(Gilad): It'd probably be better to refactor this one as it modifies the m_fYNext variable which is part of the PlayerBase
+// class and when the packet is sent to the game server, it reads the next Y coordinates to climb the terrain height.
+// Why? Because we just checking if collision detected and it should just check if collision detected and not modify any global members
+// or pointers. It would be more partical to check the whole thing if authority = GM when more code is implemented here in the future.
+// Another option would be to send a packet to the server validate the collision through the game server and let the game server check 
+// against wall hacking.
 bool CPlayerMySelf::CheckCollision()
 {
-	float fSpeed = m_fMoveSpeedPerSec * CN3Base::s_fSecPerFrm; // ���� �����̴� �ӵ�..����ӿ� ��� ���Ǿ� ����� �ӵ��̴�. 
-	if(0 == fSpeed) return false; // �������� ����� �浹üũ ���� �ʴ´�.
+	float fSpeed = m_fMoveSpeedPerSec * CN3Base::s_fSecPerFrm; // The speed at which the current moving speed is calculated.
+	if(0 == fSpeed) return false; // If it does not move, do not check collision.
 
 	__Vector3 vPos = this->Position();
 
-	__Vector3 vDir(0,0,1); // ȸ��� ���ϰ�..
+	__Vector3 vDir(0,0,1); // Find the rotation value ...
 	__Matrix44 mtxRot = this->Rotation(); 
-	vDir *= mtxRot; // ȸ�� �� ����� ���ϰ�.
+	vDir *= mtxRot; // Find the direction of rotation.
 	if(fSpeed < 0)
 	{
 		fSpeed *= -1.0f;
 		vDir *= -1.0f;
 	}
 	__Vector3 vPosNext = vPos + (vDir * fSpeed); // ��� �ġ ���..
-	if ( false == ACT_WORLD->IsInTerrainWithTerrain(vPosNext.x, vPosNext.z, vPos) )
+	if (s_pPlayer->m_InfoBase.iAuthority != AUTHORITY_MANAGER && 
```

Comment:
s_pPlayer is (the only instance of) CPlayerMySelf, which we're in right now (CPlayerMySelf::CheckCollision()). We can just access m_InfoBase directly :)